packages/pwa-kit-cli/src/ssr/server/build-dev-server.js

```diff
@@ -173,6 +146,30 @@ export const DevServerMixin = {
         })
     },
 
+    serveServiceWorker(req, res) {
+        req.app.__devMiddleware.waitUntilValid(() => {
+            const sourceMap = req.path.endsWith('.map')
+            const file = sourceMap ? 'worker.js.map' : 'worker.js'
+            const type = sourceMap ? '.js.map' : '.js'
+            const content = DevServerFactory._getWebpackAsset(req, 'client-optional', file)
+            if (content === null) {
+                res.sendStatus(404)
```

Comment:
Iirc, waitUntilValid should pause the request until webpack is able to serve the service worker. Wouldn't returning a 404 undermine that?

Reply:
No. If `content === null` then the user hasn't created a service worker. There is nothing sensible that you _could_ serve in its place, even if you wanted to.

Reminder that service workers are optional for MRT apps, but very important for Retail React App.